### PR TITLE
show budge for unread message count

### LINKED ISSUE #134: Indicator of channels with unread messages
Related to #126 and #97 it would be useful to have an indicator that there are unread messages in other channels.  As well as a count on how many mentions.   

This is what we have on our web view and our cordova app:

image

### PR DESCRIPTION
**Pull request by YusukeIwaki**
@RocketChat/android

Closes #134 

add indicator into the burger icon for notifying the count of unread room subscriptions.

Rocket.Chat (webapp) originally shows "999+" badge when the number of mentions > 999. However "999+" cannot be layout on a small badge. So "99+" is used instead in this app.

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
On just testing this one, it looks like its incrementing the number based on number of unread channels?  We might want to adjust this though to have the count only present for number of mentions / direct messages.  

Because mentioned are generally the ones a user is going to care about.  Like for instance on the demo server.  I have 8 channels with un-read messages in it.  Most of those channels I view and attack with purpose when I have the extra time to properly pay attention and help users out.  But if I know I have a direct mention or message.  I will jump right to it and knowing that at a glance is very useful.

**Comment by YusukeIwaki** [comment]
@geekgonecrazy Thank you for your feedback!
I've just understand which number would be shown on the budge.

I will adjust the behavior.

**Comment by laggedHero** [comment]
@geekgonecrazy Does this update works as expected?

**Comment by geekgonecrazy** [comment]
@laggedHero checking now!

**Comment by geekgonecrazy** [comment]
@laggedHero appears to! :+1:

File: rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/RoomToolbar.java [hunk 1]
```diff
@@ -95,4 +104,68 @@ public void setRoomIcon(@DrawableRes int drawableResId) {
...
+      try {
+        Field field = Toolbar.class.getDeclaredField("mNavButtonView");
+        field.setAccessible(true);
+        ImageButton navButtonView = (ImageButton) field.get(this);
+        int iconLeft = navButtonView.getLeft();
+        int iconTop = navButtonView.getTop();
+        int iconRight = navButtonView.getRight();
+        int iconBottom = navButtonView.getBottom();
+
+        int budgeLeft = iconLeft + (iconRight - iconLeft) * 5 / 8;
```
Comment (laggedHero) [inline rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/RoomToolbar.java]: Could you please add some explanation for these maths? Thx

Comment (YusukeIwaki) [reply rocket-chat-android-widgets/src/main/java/chat/rocket/android/widget/RoomToolbar.java]: Sure. And I will also fix typo `budge` 😓

**Review by laggedHero — APPROVED** [review]

**Pull request merged**